### ISSUE TITLE
Turn on `Smart` mode for `Blank line is statement delimiter` by default

### ISSUE THREAD

**Issue by ask9** [dbeaver/dbeaver#34422]
### Description

If select-query contains empty line and user try to execute it using Ctrl + Enter, it gives syntax error.

![0000002686](https://github.com/dbeaver/dbeaver/assets/33929221/cc6122e5-b1a8-488f-9e55-af15c1181e51)

### DBeaver Version

Community Edition 24.1.0

### Operating System

Windows 10

### Database and driver

PostgreSQL 16
PostgreSQL JDBC Driver 42.7.2

### Steps to reproduce

1. Use this code:
```
select 
    p.relname,
    
    p.relnamespace
from pg_class as p;
```

2. set caret to some place inside the query and execute it using hotkey Ctrl + Enter

Expected: query executed successfully (no syntax error)
Actual: query gives syntax error

### Additional context

Boundaries of the query should be detected by symbol ; (or maybe using some other way), but not by empty lines.
For comparison, DbForge has no such a limitation in the same test case scenario.
It means it is possible to detect query boundaries of a current sql-statement near caret correctly.

Why it is important?
- Empty lines help to manage code and keep its readability.

**Comment by E1izabeth** [comment]
You can change `Blank line is statement delimiter` setting to `Smart` to allow DBeaver rely on empty lines, but check, if it's a separator in a concrete case or not.
or to `Never` to never rely on empty lines, while detecting query boundaries

![image](https://github.com/dbeaver/dbeaver/assets/28875055/777c14f5-cecb-4298-a7b3-b9d3bd277e7a)

**Comment by ask9** [comment]
Thanks! 
Perhaps you should consider setting the default value to Smart or Never for new installations, right?

**Comment by E1izabeth** [comment]
Yes, we plan it. We added this `Smart` mode recently specifically for cases like you provided.
